src/libcollectdclient/network_parse.c

```diff
@@ -114,11 +107,11 @@ static int buffer_next(buffer_t *b, void *out, size_t n) {
 }
 
 static int buffer_uint16(buffer_t *b, uint16_t *out) {
-  uint16_t tmp;
+  char tmp[2];
   if (buffer_next(b, &tmp, sizeof(tmp)) != 0)
     return -1;
 
-  *out = be16toh(tmp);
+  *out = *(uint16_t *)tmp;
```

Comment:
That doesn't work for little endian machines. You need this snippet from the blog post:

```c
i = (data[3]<<0) | (data[2]<<8) | (data[1]<<16) | (data[0]<<24);
```

Adapted to 16 bit, you'll end up with:

```c
*out = (tmp[0] << 8) | (tmp[1] << 0);
```